src/sync.js

```diff
@@ -431,30 +440,43 @@ class Sync {
           log.error('Invalid move')
           log.error(from)
           log.error(doc)
-          side.addFolder(doc, function (err) {
-            if (err) { log.error(err) }
-            side.trash(from, function (err) {
-              if (err) { log.error(err) }
-              callback(new Error('Invalid move'))
-            })
-          })
+          try {
+            await side.addFolderAsync(doc)
+          } catch (err) {
+            log.error(err)
+            try {
+              await side.trashAsync(from)
```

Comment:
Worse is that `folderChanged()` tests are actually passing. I'll need to make sure they don't.